python/semantic_kernel/planning/stepwise_planner/stepwise_planner.py

```diff
@@ -0,0 +1,459 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import itertools
+import json
+import os
+import re
+from typing import TYPE_CHECKING, Dict, List
+
+from semantic_kernel.kernel import Kernel
+from semantic_kernel.orchestration.sk_context import SKContext
+from semantic_kernel.planning.plan import Plan
+from semantic_kernel.planning.planning_exception import PlanningException
+from semantic_kernel.planning.stepwise_planner.stepwise_planner_config import (
+    StepwisePlannerConfig,
+)
+from semantic_kernel.planning.stepwise_planner.system_step import SystemStep
+from semantic_kernel.semantic_functions.prompt_template import PromptTemplate
+from semantic_kernel.semantic_functions.prompt_template_config import (
+    PromptTemplateConfig,
+)
+from semantic_kernel.semantic_functions.semantic_function_config import (
+    SemanticFunctionConfig,
+)
+from semantic_kernel.skill_definition.function_view import FunctionView
+from semantic_kernel.skill_definition.sk_function_context_parameter_decorator import (
+    sk_function_context_parameter,
+)
+from semantic_kernel.skill_definition.sk_function_decorator import sk_function
+
+if TYPE_CHECKING:
+    from semantic_kernel.orchestration.sk_function_base import SKFunctionBase
+
+
+CUR_DIR = os.path.dirname(os.path.realpath(__file__))
+PROMPT_CONFIG_FILE_PATH = os.path.join(CUR_DIR, "Skills/StepwiseStep/config.json")
+PROMPT_TEMPLATE_FILE_PATH = os.path.join(CUR_DIR, "Skills/StepwiseStep/skprompt.txt")
+
+
+def read_file(file_path: str) -> str:
+    with open(file_path, "r") as file:
+        return file.read()
+
+
+# TODO: Original C# uses "StepwisePlanner_Excluded" for RESTRICTED_SKILL_NAME
+RESTRICTED_SKILL_NAME = "StepwisePlanner"
+S_FINAL_ANSWER_REGEX = re.compile(r"\[FINAL ANSWER\](?P<final_answer>.+)", re.DOTALL)
```

Comment:
I've noted some models having trouble here at times with this being multi word label. Sometimes it produces [FINAL_ANSWER] or [FINAL-ANSWER] and [FINALANSWER]. Probably could make it even more robust to parsing issues.

```suggestion
S_FINAL_ANSWER_REGEX = re.compile(r"\[FINAL[_\s\-]ANSWER\](?P<final_answer>.+)", re.DOTALL)
```